RecoEgamma/EgammaElectronProducers/plugins/alpaka/PixelMatchingAlgo.dev.cc

```diff
@@ -0,0 +1,308 @@
+				printf("Calculate the magnetic field with the parabolic approximation at the SC position : %f\n",magneticFieldParabolicPortable::magneticFieldAtPoint(acc, position));			
+			}
+#endif
+		}
+	};
+
+   	//--- Kernel for printing the electron seeds SoA
+	class printElectronSeedSoA {
+	public:
+	template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+	ALPAKA_FN_ACC void operator()(TAcc const& acc,
+									reco::EleSeedDeviceCollection::View view,
+									int32_t size) const 
+		{
+
+			for (int32_t i : uniform_elements(acc, size)) 
+			{
+
+				if(i>=size)
+    				break;
+
+				auto seed = view[i];
+
+				printf("For electron seed %d number of hits is %d \n ",i,seed.nHits());
+				printf("Seed id: %d is matched? %d and the matchesSC id %d \n",seed.id(),seed.isMatched(),seed.matchedScID());
+				printf("Hit 0 is valid: %d\n",seed.hit0isValid());
+				printf("Hit 0 detId is : %d\n",seed.hit0detectorID());
+				printf("Hit 1 is valid: %d\n",seed.hit1isValid());
+				printf("Hit 1 detId is : %d\n",seed.hit1detectorID());
+				printf("Hit 0 position is (%lf , %lf , %lf ) \n", seed.hit0Pos().x(),seed.hit0Pos().y(),seed.hit0Pos().z());
+				printf("Hit 0 surface is (%lf , %lf , %lf ) \n", seed.surf0Pos().x(),seed.surf0Pos().y(),seed.surf0Pos().z());
+				printf("Hit 0 rotation is (%lf , %lf , %lf ) \n", seed.surf0Rot().x(),seed.surf0Rot().y(),seed.surf0Rot().z());
+				printf("Hit 1 position is (%lf , %lf , %lf ) \n", seed.hit1Pos().x(),seed.hit1Pos().y(),seed.hit1Pos().z());
+				printf("Hit 1 surface is (%lf , %lf , %lf ) \n", seed.surf1Pos().x(),seed.surf1Pos().y(),seed.surf1Pos().z());
+				printf("Hit 1 rotation is (%lf , %lf , %lf ) \n", seed.surf1Rot().x(),seed.surf1Rot().y(),seed.surf1Rot().z());
+				if(seed.nHits()>2){
+					printf("Hit 2 is valid: %d",seed.hit2isValid());
+					printf("Hit 2 detId is : %d",seed.hit2detectorID());
+					printf("Hit 2 position is (%lf , %lf , %lf ) \n ", seed.hit2Pos().x(),seed.hit2Pos().y(),seed.hit2Pos().z());
+					printf("Hit 2 surface is (%lf , %lf , %lf ) \n", seed.surf2Pos().x(),seed.surf2Pos().y(),seed.surf2Pos().z());
+					printf("Hit 2 rotation is (%lf , %lf , %lf ) \n", seed.surf2Rot().x(),seed.surf2Rot().y(),seed.surf2Rot().z());
+				}
+			}
+		}
+	};
+
+	//--- Kernel for performing the seed to SC match
+	//--- Still dummy and just testing out some things
+	class SeedToSuperClusterMatcher {
+	public:
+		template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+		ALPAKA_FN_ACC void operator()(TAcc const& acc,
+					      reco::EleSeedDeviceCollection::View viewEleSeeds,
+					      const int32_t sizeEleSeeds,
+					      reco::SuperclusterDeviceCollection::View viewSCs,
+					      const int32_t sizeSCs,
+					      const double vtx_x,
+					      const double vtx_y,
+					      const double vtx_z) const 
+		{
+
+			const Vec3d vertex(vtx_x,vtx_y,vtx_z); 
+
+			for (int i : uniform_elements(acc,sizeEleSeeds)) 
```

Comment:
The `view.metadata().size()` is available on the device as well, so this could be
```suggestion
			for (int i : uniform_elements(acc, viewEleSeeds.metadata.size())) 
```
That will code an assumption of the `viewEleSeeds` always having exactly the right number of elements (i.e. it won't be overallocated). It is a good assumption when it is filled on the host and copied to device (because the exact number of elements is known).